### PR TITLE
Feat/ROE-2637: Retrieve application data from DB for the presenter page

### PR DESCRIPTION
**Pull request by mwejuli-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2637

### Change description

- Retrieves application data from the API for the presenter page
- Refactors the `setApplicationData` method to update API with new application data
- Modifies tests to match new/updated functionality
- Includes minor formatting changes

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### CODE REVIEW

File: src/utils/application.data.ts [hunk 1]
```diff
@@ -88,18 +99,42 @@ export const deleteApplicationData = (session: Session | undefined): boolean | u
...
+  let session: Session | undefined;
 
-  if (ApplicationDataArrayType.includes(key)){
-    if ( !appData[key] ) { appData[key] = []; }
-    appData[key].push(data);
-  } else {
-    appData = { ...appData, [key]: { ...data } } as ApplicationData;
+  try {
+
+    if (!isActiveFeature(FEATURE_FLAG_ENABLE_REDIS_REMOVAL) || (sessionOrRequest instanceof Session)) {
```
Comment (markpit) [inline src/utils/application.data.ts:110]: i think we might not need the flag check here, getApplicationData can take either session or request and we seem to call the same function in the if and else. Can't we just call getApplicationData with sessionOrRequest ?

Comment (markpit) [reply src/utils/application.data.ts:110]: ok, we can leave it like this.

Comment (mwejuli-ch) [reply src/utils/application.data.ts:110]: cheers

File: src/utils/application.data.ts [hunk 2]
```diff
@@ -88,18 +99,42 @@ export const deleteApplicationData = (session: Session | undefined): boolean | u
...
+      }
+      appData[key].push(data);
+    }
+
+    if (session) {
+      return setExtraData(session, appData);
+    }
+
+    setExtraData(req?.session as Session, appData);
+    return updateOverseasEntity(req as Request, req?.session as Session, appData);
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/utils/application.data.ts:132]: oh i see it returns above if it is a session

File: test/controllers/presenter.controller.spec.ts [hunk 3]
```diff
@@ -105,15 +134,16 @@ describe("PRESENTER controller", () => {
...
+      mockGetApplicationData.mockReturnValue({ [PresenterKey]: PRESENTER_OBJECT_MOCK });
       const resp = await request(app).get(PRESENTER_URL);
 
       expect(resp.status).toEqual(200);
       expect(resp.text).toContain(OVERSEAS_NAME_URL);
     });
 
     test("catch error when renders the presenter page", async () => {
-      mockGetApplicationData.mockImplementationOnce( () => { throw new Error(ANY_MESSAGE_ERROR); });
+      // mockGetApplicationData.mockImplementationOnce( () => { throw new Error(ANY_MESSAGE_ERROR); });
```
Comment (markpit) [inline test/controllers/presenter.controller.spec.ts:145]: comment can be removed

File: test/controllers/presenter.controller.spec.ts [hunk 4]
```diff
@@ -149,7 +179,8 @@ describe("PRESENTER controller", () => {
...
```
Comment (markpit) [inline test/controllers/presenter.controller.spec.ts:182]: comment

File: test/controllers/secure.register.filter.controller.spec.ts [hunk 5]
```diff
@@ -87,7 +96,8 @@ describe( "SECURE REGISTER FILTER controller", () => {
   describe("GET tests", () => {
 
     test(`renders the ${config.SECURE_REGISTER_FILTER_PAGE} page`, async () => {
-      mockGetApplicationData.mockReturnValueOnce({});
+      // mockGetApplicationData.mockReturnValueOnce({});
```
Comment (markpit) [inline test/controllers/secure.register.filter.controller.spec.ts:99]: a few of these commented code can be removed

File: test/controllers/presenter.controller.spec.ts (hunk 3 above)
Comment (markpit) [reply test/controllers/presenter.controller.spec.ts:145]: yeah, i would remove them in case we never get round to doing the rest of the work!

Comment (mwejuli-ch) [reply test/controllers/presenter.controller.spec.ts]: sure, all removed now.

File: test/controllers/presenter.controller.spec.ts (hunk 4 above)
Comment (mwejuli-ch) [reply test/controllers/presenter.controller.spec.ts]: removed

File: test/controllers/secure.register.filter.controller.spec.ts (hunk 5 above)
Comment (mwejuli-ch) [reply test/controllers/secure.register.filter.controller.spec.ts]: removed

File: test/controllers/update/secure.update.filter.controller.spec.ts [hunk 6]
```diff
@@ -84,7 +89,8 @@ describe("SECURE UPDATE FILTER controller", () => {
 
     test(`renders the ${SECURE_UPDATE_FILTER_PAGE} page for remove`, async () => {
       mockIsActiveFeature.mockReturnValueOnce(false);
-      mockGetApplicationData.mockReturnValueOnce({});
+      // mockGetApplicationData.mockReturnValueOnce({});
```
[the remark this answers is not in the document]
Comment (mwejuli-ch) [reply test/controllers/update/secure.update.filter.controller.spec.ts]: removed

**Review by markpit — APPROVED** [review]

**Review by lduranteau — APPROVED** [review]

**Pull request merged**